NSURL.m

```diff
@@ -351,7 +364,28 @@ static CFTypeRef CFURLCreatePropertyForKey(CFURLRef url, CFStringRef key, CFErro
     else if (CFEqual(key, kCFURLTypeIdentifierKey))
     {
         // Key for the resource’s uniform type identifier (UTI), returned as a CFString object.
+        void* LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
+        if(LaunchServicesHandle == NULL){
+            dlError(error);
+            return NULL;
+        }
+        CFStringRef (*UTTypeCreatePreferredIdentifierForTag)(CFStringRef, CFStringRef, _Nullable CFStringRef) = dlsym(LaunchServicesHandle, "UTTypeCreatePreferredIdentifierForTag");
+        CFStringRef *tagType = dlsym(LaunchServicesHandle, "kUTTagClassFilenameExtension");
+        if(UTTypeCreatePreferredIdentifierForTag == NULL || tagType == NULL){
+            dlclose(LaunchServicesHandle);
+            dlError(error);
```

Comment:
Do `dlError` first, otherwise the call to `dlclose` could overwrite the stored error.